lib/group/group-router.js

```
@@ -1,56 +1,66 @@
 'use strict';
 
 var express = require('express'),
-  config = require('./config-group');
+  config = require('./config-group'),
+  shortid = require('shortid');
 
-function initRouter(mediator) {
+var GroupCloudDataTopics = require('fh-wfm-mediator/lib/topics');
+
+/**
+ * Function which handles the response according to the promise returned by the topic
+ *
+ * @param {Object} res
+ * @param {Promise} topicPromise Promise returned by the topic the request was sent to
+ */
+function handleResponse(res, topicPromise) {
+  topicPromise.then(function(data) {
+    res.status(200).json(data);
+  }).catch(function(err) {
+    console.log("[Error] Group topic request: ", err);
+    res.status(500).json({error: err.message});
+  });
+}
```

This one should be moved somewhere where it's more reusable, else we'll have a copy of it on a lot of routers.

Also `.catch(function(err) {throw err;})` instead of finishing the response so it can get handled by the global express error handler.

This is also a nice opportunity to think with higher-order functions. How would this have to look like if the usage was:
```javascript
router.route('/').get(handlePromiseResponse(function(req, res) {
  return Promise.resolve({some: 'data'}); // and with `return {some: 'data'};`
}));
```

---

I'm actually not sure about the catch suggestion. Any error thrown inside any promise chain is implicitly wrapped in a try/catch, so this error will just be propagated along as if this catch weren't even there. Unless Express is listening to the uncaught promise exception handler, it would never see that error. (I think there are Bluebird configurations for this.)

---

Ah, yes, just tested this and they do swallow all the exceptions. I guess the way to go then is
```javascript
(req, res, next) => {
// ...
  .catch(next)
}
```

There's no difference between 2-arity and 3-arity handlers btw, you can still just not call `next` if you're ending the response. Only 4-arity is special-cased for error handling.

---

https://gist.github.com/paolobueno/61793824d22de69823d9d459adf6d4b5/revisions